[RM Ch3] Update SFC Section 3.5.1 (#2309)
cntt-n/CNTT: doc/ref_model/chapters/chapter03.md

There is also a need to provide specialised tools to aid troubleshooting of individual services and the communication between them in order to investigate issues in the performance of composed network functions. Minimally, there is a need to provide packet level and byte level counters and statistics as the packets pass through the service chain in order to ascertain any issues with forwarding and performance. Additionally, there is a need for mechanisms to trace the paths of selected subsets of traffic as they flow through the service chain.

<a name="3.5.1.1"></a>

> #### 3.5.1.1 Introduction 

Reviewer: This is more or less redundant to the text in chapter 3.5.1. Can you please merge these two?

Author: @CsatariGergely removed the line mentioning acyclic graph as it is already part of 3.5.1

Reviewer: I would still merge the two chapters as both of them are some kind of introduction.

Author: @CsatariGergely  i feel introduction is bad title for sub section .. can we rename it as "SFC layering model " rather than introduction ? will it make it more specific or still we merge it with previous section?

> #### 3.5.1.1 Introduction 

Reviewer 2: ```suggestion
#### 3.5.1.1 Service Function Chaining Model Introduction 
```

> Service function chaining (SFC) can be visualized as layered structure where Service function plane (SFC data plane, consists of service function forwarder, classifier, service function, service function proxy) resides over Service function overlay network. 

Reviewer 2: ```suggestion
Service Function Chaining (SFC) can be visualized as a layered structure where the Service Function plane (SFC data plane, consists of service function forwarder, classifier, service function, service function proxy) resides over a Service Function overlay network. 
```

> Service function chaining (SFC) can be visualized as layered structure where Service function plane (SFC data plane ,consists of Service function forwarder, Classifier, Service function, Service function proxy) resides over Service function overlay network. At a very high level, a service function plane is a directed acyclic graph with the composing Servie functions being the vertices.  

Reviewer: nit: "plane ,consists" -> "plane, consists"

Author: @CsatariGergely Accepted and will change.

> Service function chaining (SFC) can be visualized as layered structure where Service function plane (SFC data plane ,consists of Service function forwarder, Classifier, Service function, Service function proxy) resides over Service function overlay network. At a very high level, a service function plane is a directed acyclic graph with the composing Servie functions being the vertices.  

Reviewer: Classifier, Service function and Service function proxy are not defined in the Glossary so they should not be capitalized.

Author: @CsatariGergely Sure will align .

> Service function chaining (SFC) can be visualized as layered structure where Service function plane (SFC data plane ,consists of Service function forwarder, Classifier, Service function, Service function proxy) resides over Service function overlay network. At a very high level, a service function plane is a directed acyclic graph with the composing Servie functions being the vertices.  

Reviewer: If this service function plane and overlay network a description of a concrete SFC solution or these are industry standard components of SFC?

Author: @CsatariGergely these are not industry standard but visualization -
i tried to rephrase rfc 8294 section 2 service layering model  ( https://tools.ietf.org/html/rfc8924). 
because it will be easier to visualize these way and define model. i will see if i can rephrase better.

> Service function chaining utilizes a service-specific overlay that creates the service topology.  The service overlay provides service function connectivity, built "on top" of the existing network topology. It leverages various overlay network technologies (e.g., Virtual eXtensible Local Area Network (VXLAN)) for interconnecting SFC data-plane elements and allows establishing Service Function Paths (SFPs).

Reviewer 2: ```suggestion
SFC utilizes a service-specific overlay that creates the service topology.  The service overlay provides service function connectivity built "on top" of the existing network topology. It leverages various overlay network technologies (e.g., Virtual eXtensible Local Area Network (VXLAN)) for interconnecting SFC data-plane elements and allows establishing Service Function Paths (SFPs).
```

> In Overlay network, packets are routed based on networking principles as use a suitable path for the packet from source to its destination in network. 

Reviewer 2: ```suggestion
In an overlay network, packets are routed based on networking principles and use a suitable path for the packet to be routed from a source to its destination. 
```

> However, in service overlay network, packets are routed based on policies unlike overlay network, again defined at Orchestrator level. This requires specific support at network level such as  at CNI in CNF environment to provide such specific routing mechanism.

Reviewer 2: ```suggestion
However, in a service-specific overlay network, packets are routed based on policies. This requires specific support at network level such as  at CNI in CNF environment to provide such specific routing mechanism.
```
